proto/neutron/interchainqueries/tx.proto

```diff
@@ -13,147 +13,194 @@ import "tendermint/crypto/proof.proto";
   string sender = 6;
 }
 
+// Response type for the Msg/RegisterInterchainQuery RPC method.
 message MsgRegisterInterchainQueryResponse {
+  // The ID assigned to the registered Interchain Query by the module.
   uint64 id = 1;
 }
 
+// Request type for the Msg/SubmitQueryResult RPC method.
 message MsgSubmitQueryResult {
   option (cosmos.msg.v1.signer) = "sender";
-
+  // The ID of the Interchain Query.
   uint64 query_id = 1;
+  // The signer of the message.
   string sender = 2;
-
-  // is the IBC client ID for an IBC connection between Neutron chain and target
-  // chain (where the result was obtained from)
+  // The IBC client ID that corresponds to the IBC connection to the remote chain (where the
+  // query result is coming from).
   // Deprecated: populating this field does not make any affect
   string client_id = 3 [deprecated = true];
+  // The result of the Interchain Query execution.
   QueryResult result = 4;
 }
 
+// Contains different information about a single Interchain Query execution result. Currently,
+// this structure is used both in query result submission via an ICQ Relayer and as a query result
+// storage for read/write operations to interchainqueries module, but the structure fields are
+// populated in a bit different ways. When submitting a query result, all fields are populated and
+// provided to the interchainqueries module in order to verify the result against the IBC client's
+// state. But in order to lighten the chain state, the interchainqueries module removes the block
+// field and proofs from the kv_results.
 message QueryResult {
+  // A list of a KV Interchain Query execution results. Each result contains query parameters, a
+  // response value and a proof.
   repeated StorageValue kv_results = 1;
+  // A TX Interchain Query execution result. Contains metainformation about the blocks of the query
+  // execution height. Only populated when submitting an Interchain Query result for verification
+  // and emptied when saving the result on chain.
   Block block = 2;
+  // The height of the chain at the moment of the Interchain Query execution.
   uint64 height = 3;
+  // The revision number of the chain at the moment of the Interchain Query execution.
   uint64 revision = 4;
+  // Whether to send the query result to the owner contract as a sudo message. Only applicable for
+  // KV type of Interchain Queries.
   bool allow_kv_callbacks = 5;
 }
 
+// A verifiable result of performing a single KVKey read.
 message StorageValue {
-  // is the substore name (acc, staking, etc.)
+  // The first half of the storage path. It is supposed to be a substore name for the query
+  // (e.g. bank, staking, etc.).
   string storage_prefix = 1;
-
-  // is the key in IAVL store
+  // The second half of the storage path. The remaining part of the full path to an IAVL storage node.
   bytes key = 2;
-
-  // is the value in IAVL store
+  // A base64-encoded value read from the given storage path.
```

Comment:
This is a bit misleading i think.
The value is not base64-encoded, it's just `bytes` field. Base64 encoding happens when you try to convert this into some human-readable format (JSON, etc). In that way `bytes` fields will be represented as base64 string field.

But your comments implies that the `value` field itself is base64 encoded + it has `bytes` type